arff.py

```diff
@@ -610,11 +633,20 @@ def _decode_attribute(self, s):
 
         else:
             # If not nominal, verify the type name
-            type_ = unicode(type_).upper()
-            if type_ not in ['NUMERIC', 'REAL', 'INTEGER', 'STRING']:
+            utype_ = unicode(type_).upper()  # if type DATE, we don't want to upper() any format str
+            if 'DATE' in type_ and ' ' in type_:
```

Comment:
Could you please add a regular expression `_RE_TYPE_DATE` to follow the current style? Also, the if 'DATE' in check should then be moved one level out and become an elif statement.

Reply:
Why a regular expression? How does line 637 not solve the problem simply, elegantly and flexibly?

Reply:
It does, I'm just trying to keep the package consistent. You can also leave it for now as this is only cosmetic.